pkg/operator/statefulset.go

```diff
@@ -17,98 +17,194 @@ package operator
 import (
 	"fmt"
 	"net/url"
+	"strconv"
 
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 const configReloaderPort = 8080
 
+type ConfigReloader struct {
```

Comment:
Shall we just rename the file from `statefulset.go` to `config-reloader.go`,  but kept under the same `pkg/operator` as other files that are used for composing operator instead of creating another `pkg`?

Will add godocs as per suggested :-)

Reply:
> Shall we just rename the file from statefulset.go to config-reloader.go, but kept under the same pkg/operator as other files that are used for composing operator instead of creating another pkg?

My understanding is that config-reloader is not used for composing the operator, but rather as a sidecar for prometheus and alertmanager. You can see that the `CreateConfigReloader` function is referenced from the alertmanager and prometheus packages which creates a bit of a circular dependency. 

In any case, I would not mind just renaming the file to `config-reloader.go` and leaving it in the same package since that would already be an improvement. I leave the decision up to you depending on how much effort you think each approach will require :)

Reply:
You are right; I should not have used the wording `composing the operator` and apologize for that!

Yah, `CreateCofnigReloader` is called in funcs named `makeStatefulSetSpec` in files named `statefulset.go` under `pkg/prometheus` , `pkg/alertmanager` and `pkg/thanos`...which **might** be  the reason why the file name is ALSO called `statefulset.go` under `pkg/operator`...LoL

As funcs in other files, such as `image.go` under `pkg/operator` callled by `makeStatefulSetSpec`; renaming seams the minial effort to take for improvment.